tools/identity-resolution/identity-resolution/Helpers/GitHubAadConverter.cs

```diff
@@ -0,0 +1,71 @@
+using System;
+using System.Net.Http;
+using Azure.Sdk.Tools.NotificationConfiguration.Models;
+using Microsoft.Extensions.Logging;
+using Newtonsoft.Json;
+
+namespace Azure.Sdk.Tools.NotificationConfiguration.Helpers
+{
+    public class GitHubToAadConverter
+    {
+        /// <summary>
+        /// GitHubToAadConverter constructor for generating new token, and initialize http client.
+        /// </summary>
+        /// <param name="clientId">AAD Client/App Id</param>
+        /// <param name="clientSecret">AAD Client/App Secret</param>
+        /// <param name="tenantID">AAD Tenant Id</param>
+        /// <param name="logger">Logger</param>
+        public GitHubToAadConverter(
+            string clientId,
+            string clientSecret,
+            string tenantID,
+            ILogger<GitHubNameResolver> logger)
+        {
+            try
+            {
+                opsAuthToken = new AadTokenCredential(clientId, clientSecret, tenantID).GenerateAadToken();
+            }
+            catch (Exception ex)
+            {
+                logger.LogError("Failed to generate aad token. " + ex.Message);
+            }
+            InitializeOpenSourceClient(opsAuthToken);
+            this.logger = logger;
+        }
+
+        private readonly string opsAuthToken;
```

Comment:
Do we need to keep this value as a field?

Reply:
Removed.